cloud/parameterstore/ssm_client.go

```diff
@@ -114,19 +113,59 @@ func (c *ssmClientImpl) GetParametersSimple(ctx context.Context, input *ssm.GetP
 }
 
 func (c *ssmClientImpl) GetParameters(ctx context.Context, input *ssm.GetParametersInput) ([]*ssm.GetParametersOutput, error) {
-	// TODO (DEVPROD-9391): follow-up PR should handle batching requests due
-	// to limit of 10 parameters per API call.
-	output, err := retrySSMClientOp(ctx, "GetParameters", c.client, input, func(ctx context.Context, client *ssm.Client, input *ssm.GetParametersInput) (*ssm.GetParametersOutput, error) {
+	const maxParamsPerRequest = 10
+	return retryBatchSSMClientOp(ctx, input.Names, maxParamsPerRequest, c.client, input, func(names []string, input *ssm.GetParametersInput) *ssm.GetParametersInput {
+		batchInput := *input
+		batchInput.Names = names
+		return &batchInput
+	}, func(ctx context.Context, client *ssm.Client, input *ssm.GetParametersInput) (*ssm.GetParametersOutput, error) {
 		return client.GetParameters(ctx, input)
-	})
-	if err != nil {
-		return nil, err
+	}, "GetParameters")
+}
+
+// retryBatchSSMClientOp runs an SSM operation with retries and handles batching for
+// SSM API methods that have a limit on the number of parameters that can be
+// queried at once.
+func retryBatchSSMClientOp[Input interface{}, Output interface{}](ctx context.Context, names []string, maxParamsPerRequest int, client *ssm.Client, input Input, makeBatchInput func(names []string, input Input) Input, op func(ctx context.Context, client *ssm.Client, input Input) (Output, error), opName string) ([]Output, error) {
+	batches := makeParamNameBatches(names, maxParamsPerRequest)
+
+	var outputs []Output
+	for _, batch := range batches {
+		batchInput := makeBatchInput(batch, input)
+		output, err := retrySSMClientOp(ctx, client, batchInput, op, opName)
+		if err != nil {
+			return nil, err
+		}
+		outputs = append(outputs, output)
+	}
+	return outputs, nil
+}
+
+// makeParamNameBatches partitions names into batches with at most
+// maxParamsPerBatch in each batch. This is to handle batching SSM requests that
+// limit the number of parameters that can be queried/modified in a single API
+// call.
+func makeParamNameBatches(paramNames []string, maxParamsPerBatch int) [][]string {
```

Comment:
Since this function is ultimately just string / array manipulation can we move it to the util package in case it's helpful for something else in the future?

Reply:
Sure, I opened a PR: https://github.com/evergreen-ci/utility/pull/68